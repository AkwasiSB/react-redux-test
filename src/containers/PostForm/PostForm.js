import PropTypes from 'prop-types';
import React, { useState } from 'react';
import { connect } from 'react-redux';
import { handleCreateNewPost } from '../../redux/actions/postActions';

const PostForm = function(props) {
	const [title, setTitle] = useState('');
	const [body, setBody] = useState('');
	const formRef = React.createRef();

	function handleTitleChange(e) {
		setTitle(e.target.value);
	}

	function handleBodyChange(e) {
		setBody(e.target.value);
	}

	function handleFormSubmit(e) {
		e.preventDefault();

		const post = {
			title,
			body
		};

		setTitle('');
		setBody('');

		props.createPost(post);
	}

	return (
		<form ref={formRef} onSubmit={handleFormSubmit}>
			<div>
				<label>Title: </label>
				<input type='text' value={title} onChange={handleTitleChange} />
			</div>

			<div>
				<label>Body: </label>
				<textarea value={body} onChange={handleBodyChange} />
			</div>
			<br />
			<button type='submit'>Submit</button>
		</form>
	);
};

PostForm.propTypes = {
	createPost: PropTypes.func.isRequired,
};

function mapDispatchToProps(dispatch) {
	return {
		createPost: (post) => dispatch(handleCreateNewPost(post))
	};
}

export default connect(null, mapDispatchToProps)(PostForm);
