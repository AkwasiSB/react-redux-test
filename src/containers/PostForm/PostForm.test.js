import PostForm from './PostForm';

test('if PostForm renders correctly', () => {
    const postFormWrapper = shallow(<PostForm createPost={() => {}} />);

    expect(postFormWrapper).toMatchSnapshot();
});
