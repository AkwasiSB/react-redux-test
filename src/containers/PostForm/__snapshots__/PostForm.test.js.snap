// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if PostForm renders correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
